z_json_test.go

```diff
@@ -705,6 +705,126 @@ func TestUpdateJSONScalar(t *testing.T) {
 	}
 }
 
+// Converts map with different Go-types to godror.JSON type
+// This is bound to the insert function and executed
+// For each unique go-type in the map, their corresponding JSON types,
+// as stored in the DB, are fetched and compared with their expected values.
+func TestStorageTypes(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithTimeout(testContext("StorageTypes"),
+		30*time.Second)
+	defer cancel()
+	conn, err := testDb.Conn(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+	tbl := "test_objectcollection_jsonmap" + tblSuffix
+	conn.ExecContext(ctx, "DROP TABLE "+tbl)
+	_, err = conn.ExecContext(ctx,
+		"CREATE TABLE "+tbl+" (id NUMBER(6), jdoc JSON)", //nolint:gas
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf(" JSON Document table  %q: ", tbl)
+
+	defer testDb.Exec(
+		"DROP TABLE " + tbl, //nolint:gas
+	)
+
+	stmt, err := conn.PrepareContext(ctx,
+		"INSERT INTO "+tbl+" (id, jdoc) VALUES (:1, :2)", //nolint:gas
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer stmt.Close()
+	//        birthdate, err := time.Parse(time.UnixDate, "Wed Feb 25 11:06:39 PST 1990")
+	birthdate := time.Now()
```

Comment:
Maybe a constant date with controlled time zone info would test more corner cases.
You can use the global `birthdate` variable.